docs/test-process.md

```diff
@@ -0,0 +1,62 @@
+# Rancher2 Provider
+
+_For Terraform Maintainers Only_
+
+QA has a new process to test Terraform using RCs which will enable us to better test this tool before a release is published.
```

Comment:
Nice! Hopefully this new page will help prevent any confusion on what aspects of the development process are expected of community members versus maintainers